### ISSUE TITLE
option to warn on "unable to resolve prefix" for shacl rules

### ISSUE THREAD

**Issue by majidaldo** [TopQuadrant/shacl#179]
There are just too many cases where shacl sparql definitions  do not have the right `sh:prefixes`. I suggest that the processor should issue a warning instead of erroring out. It should give the location of the violation as well.

**Comment by HolgerKnublauch** [comment]
Yes, it was unfortunate that the SHACL spec is not using the prefixes from the context graph directly. Almost every user runs into that problem. In our product we have since introduced a flag that (by default) automatically uses the prefixes from the Jena graph. A similar flag could be added here to the open source version.
